api/src/test/java/org/pmiops/workbench/reporting/ReportingVerificationServiceTest.java

```diff
@@ -0,0 +1,156 @@
+import java.util.Arrays;
+import java.util.List;
+import java.util.logging.Handler;
+import java.util.logging.Logger;
+import java.util.logging.StreamHandler;
+import javax.persistence.EntityManager;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.pmiops.workbench.api.BigQueryService;
+import org.pmiops.workbench.db.dao.CdrVersionDao;
+import org.pmiops.workbench.db.dao.UserDao;
+import org.pmiops.workbench.db.dao.WorkspaceDao;
+import org.pmiops.workbench.db.jdbc.ReportingQueryServiceImpl;
+import org.pmiops.workbench.db.model.DbCdrVersion;
+import org.pmiops.workbench.db.model.DbUser;
+import org.pmiops.workbench.model.ReportingUser;
+import org.pmiops.workbench.test.FakeClock;
+import org.pmiops.workbench.testconfig.ReportingTestConfig;
+import org.pmiops.workbench.testconfig.fixtures.ReportingTestFixture;
+import org.pmiops.workbench.utils.FieldValues;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.annotation.DirtiesContext.ClassMode;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@DirtiesContext(classMode = ClassMode.BEFORE_EACH_TEST_METHOD)
+public class ReportingVerificationServiceTest {
+  private static final Instant NOW = Instant.parse("2000-01-01T00:00:00.00Z");
+  private static final Instant THEN_INSTANT = Instant.parse("1989-02-17T00:00:00.00Z");
+
+  private static final Long ACTUAL_COUNT = (long) 2;
+  private static final List<FieldValueList> ACTUAL_COUNT_QUERY_RESULT =
+      ImmutableList.of(
+          FieldValues.buildFieldValueList(
+              FieldList.of(Field.of("actual_count", LegacySQLTypeName.INTEGER)),
+              Arrays.asList(new Object[] {Long.toString(ACTUAL_COUNT)})));
+
+  private static final Logger LOGGER =
+      Logger.getLogger(
+          ReportingVerificationServiceImpl.class
+              .getName()); // matches the logger in the affected class
+  private static OutputStream logCapturingStream;
+  private static StreamHandler customLogHandler;
+
+  @Autowired
+  @Qualifier("REPORTING_USER_TEST_FIXTURE")
+  ReportingTestFixture<DbUser, ReportingUser> userFixture;
+
+  @Autowired private ReportingVerificationService reportingVerificationService;
+  @Autowired private EntityManager entityManager;
+  @Autowired private CdrVersionDao cCdrVersionDao;
+  @Autowired private WorkspaceDao workspaceDao;
+  @Autowired private UserDao userDao;
+  @MockBean private BigQueryService mockBigQueryService;
+
+  @TestConfiguration
```

Comment:
Impressive that you were able to get this all working cleanly, test is pretty nice.